src/logging.rs

```diff
@@ -0,0 +1,33 @@
+use log;
+use std::sync;
+use std::io;
+
+pub struct Logger<T:Send+io::Write> {
+    level: log::LogLevelFilter,
+    output: sync::Mutex<T>
+}
+
+impl<T> Logger<T> where T:Send+io::Write {
+    pub fn new(output: T, level: log::LogLevelFilter) -> Logger<io::LineWriter<T>> {
+        Logger { level: level,
+                 output: sync::Mutex::new(io::LineWriter::new(output)) }
+    }
+}
+
+
+impl<T> log::Log for Logger<T> where T:Send+io::Write {
+    fn enabled(&self, metadata: &log::LogMetadata) -> bool {
+        metadata.level() <= self.level
+    }
+
+    fn log(&self, record: &log::LogRecord) {
+        if !self.enabled(record.metadata()) {
+            return
+        }
+        let writer = &mut self.output.lock().unwrap();
+        writer.write(format!("{} {} {}\n",
```

Comment:
Shouldn't we include the target here so one can easily tell who is talking? On TRACE there are mio logs showing up that I think would be confusing otherwise

> Set Attribute: Reset
> Set Attribute: Foreground(Named(Foreground))
> registering with poller
> registering; token=Token(1); interests=Ready {Readable}
> set_readiness event Ready {Readable} Token(0)
> returning readiness event Ready {Readable} Token(0)
> set_readiness event Ready {} Token(0)
> registering with poller
> registering with poller
> registering; token=Token(1); interests=Ready {Readable | Writable}
> registering with poller
> registering; token=Token(1); interests=Ready {Readable}
> backspace
>